client/result/result.go

```diff
@@ -0,0 +1,352 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
```

Comment:
The only bits of `client/result` shared by `repair` and `bootstrap` is the `Options` object. So least invasive change overall would be to only extract that into `client`. 

My preference is to leave it as is currently in the PR because having `client` depend on `bootstrap` mandates no file in the top-level `bootstrap` package import client due to it introducing a circular dependency. Feels icky. But obv the bootstrapping code doesn't do that now, so not opposed to doing the least invasive change of moving `Options` out.